### DISCUSSION TITLE
Efficient way to filter by list of values

### DISCUSSION THREAD

**Discussion by elmarw** [pocketbase/pocketbase#1823] [Q&A] [answered]
Is there an efficient way to query a column with a string and match it with any value in a list.

Now im doing:

`/api/collections/reviews/records?sort=-creation_date&filter=('12345,2532525,24234'~sku)`

But when I build my own query I would do something like:

`WHERE sku IN ('12345',2532525','24234')`

Which I think is way more performant? Or do you just have to do a lot of OR statements?

**Comment by ganigeorgiev** [comment, marked as answer]
The `~` operator is one way to do it.

Alternatively, as you noted, you can combine the conditions with multiple `||` statements:

```js
?filter=(sku = "sku1" || sku = "sku2" || sku = "sku3")

// OR programmatically you can create the above expression like this (_js example_):
const skus = ["sku1", "sku2", "sku3"];
const result = await pb.collection('reviews').getList(1, 20, {
    sort: '-creation_date',
    filter: skus.map((sku) => `sku="${sku}"`).join("||"),
})
```

Performance-wise, it is difficult to say since it depends how large is your dataset and whether there are other conditions in addition to the filter, but most likely the `||` approach may be slightly better, since it will be translated directly to the SQL: `sku='sku1' OR sku='sku2' OR sku='sku3'`, while the `~` operator will result in: `'sku1,sku2,sku3' LIKE ("%" || sku || "%")`.

~Technically both should be able to use an index if it was defined on the `sku` field/column~ (_support for managing indexes from the UI will be available in the near future_); on second thought, for the `~` operator since the column value is on the right side of the LIKE condition, the query planner may end up not using the eventual `sku` index, I'm not sure, it really depends on the final query and what the query planner will decide to use, but if you are concerned about the performance, the safest bet would be to combine the skus conditions with `||` (_this approach also doesn't rely on the same sku length since each sku will be compared 1:1_).
